### PR TITLE
Mock to test mpc when rpc is congested and slow

### PR DESCRIPTION
**Pull request by ailisp**
Address #642. This PR adds a configurable proxy to the mpc node when talks to lake indexer rpc and when lake indexer talks to s3 (simulate lake being slow)

image

(Also mentioned in README), **After this PR, to run chain-signatures integration test, you will also need [toxiproxy](https://github.com/Shopify/toxiproxy). Ensure `toxiproxy-server` is in `PATH`**

### CODE REVIEW

**Review by volovyks — APPROVED** [review]
Looks awesome! And easy to use.
I want to clarify where the delays are added.
- Since it is not used in workspaces, interactions from users are not delayed.
- Indexers on all nodes are getting new signature requests with a delay
- Sandbox is receiving signature responses with a delay

File: integration-tests/chain-signatures/src/containers.rs [hunk 1]
```diff
@@ -335,11 +340,35 @@ impl<'a> LakeIndexer<'a> {
         let rpc_host_port = container.get_host_port_ipv4(Self::CONTAINER_RPC_PORT);
         let rpc_host_address = format!("http://127.0.0.1:{rpc_host_port}");
 
+        let toxi_server = async_process::Command::new("toxiproxy-server")
+            .kill_on_drop(true)
+            .spawn()
+            .with_context(|| "failed to run toxiproxy-server")?;
+        let toxi_server_address = "http://127.0.0.1:8474";
```
[the remark this answers is not in the document]
Comment (ailisp) [reply integration-tests/chain-signatures/src/containers.rs:347]: OK

**Comment by ailisp** [comment]
As discussed with Serhii, I'll also add proxy from lake indexer to s3 to simulate slow down in lake under high load. However proxied s3 (even without any latency rules) seems not working, still trying to figure it out

**Comment by ailisp** [comment]
@volovyks Now lake indexer -> s3 proxy is added. It is more complicated than expected, as toxiproxy local binary can only proxy local node to container, and toxiproxy container can only proxy container to container, so there are two toxiproxy launched, one local binary running as a subprocess, one docker container. Please take another look

**Review by volovyks — APPROVED** [review]
Looks good! Let's fix the tests and merge it.

**Comment by volovyks** [comment]
> Looks good! Let's fix the tests and merge it.

Oh, I see, that is the audit.

**Pull request merged**